pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/SourceEditorController.java

```diff
@@ -126,6 +147,57 @@ public void refreshAST() {
     }
 
 
+    public void showAuxClassPathController(DesignerRoot root) {
+        AuxClassPathController auxClassPathController = new AuxClassPathController(root);
+
+        FXMLLoader fxmlLoader = new FXMLLoader(DesignerUtil.getFxml("auxclasspath-setup-popup.fxml"));
+
+        fxmlLoader.setControllerFactory(type -> {
+            if (type == AuxClassPathController.class) {
+                return auxClassPathController;
+            } else {
+                throw new IllegalStateException("Wrong controller!");
+            }
+        });
+        try {
+            Parent root1 = fxmlLoader.load();
+
+            auxClassPathController.setAuxclasspathFiles(auxclasspathFiles.getValue());
+
+            Stage stage = new Stage();
+            stage.initOwner(root.getMainStage());
+            stage.initModality(Modality.WINDOW_MODAL);
+            stage.setScene(new Scene(root1));
+            stage.show();
+
+            auxClassPathController.setOnApply(files -> {
+                stage.close();
+                auxclasspathFiles.setValue(files);
+            });
+
+            auxClassPathController.setOnCancel(stage::close);
+
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+
+    @PersistentProperty
+    public String getAuxclasspathFiles() {
+        return auxclasspathFiles.getValue().stream().map(p -> p.getAbsolutePath()).collect(Collectors.joining(File.pathSeparator));
+    }
+
+
+    public void setAuxClassPathFiles(String files) {
```

Comment:
there's a typo here. The name of the method should be exactly `setAuxclasspathFiles` (notice the capitalisation) otherwise the persistence framework is displeased ^^